src/unread/unreadModel.js

```diff
@@ -200,6 +201,31 @@ function streamsReducer(
       return deleteMessages(state, action.messages);
     }
 
+    case EVENT_UPDATE_MESSAGE: {
+      if (
+        (action.orig_subject !== undefined && action.subject !== action.orig_subject)
+        || (action.new_stream_id !== undefined && action.new_stream_id !== action.stream_id)
+      ) {
+        const moved_ids = new Set(action.message_ids);
+        const updated_unreads = state
+          .getIn([action.stream_id, action.orig_subject ?? action.subject], Immutable.List())
+          .filter(id => moved_ids.has(id));
+
+        if (updated_unreads.size > 0) {
+          return state
```

Comment:
These conditionals (this, and the `if` above) have the feeling that the main line of the code's logic gets nested deeper and deeper as the code goes on. This pattern also has the effect that what happens in the alternative case (here, when `updated_unreads` is empty) is left in suspense for the reader until after all the other code.

I tend to prefer the "early return" pattern for arranging code flow: exceptional cases like errors and fast-path shortcuts go off to the side and then return, so that the main narrative of the code gets to continue vertically all in the same column. This also means that the condition for each exceptional case, and what we do in that case, get to stay right next to each other.

I can try demonstrating that rearrangement for this code at merge time, or after merge.